socorro/cron/jobs/featured_versions_automatic.py

```diff
@@ -0,0 +1,138 @@
+import re
+
+import requests
+
+from configman import Namespace
+from crontabber.base import BaseCronApp
+from crontabber.mixins import (
+    with_postgres_transactions,
+    with_single_postgres_transaction,
+)
+
+
+def alias_list_to_dict(input_string):
+    aliases = {}
+    for item in input_string.split(','):
+        if not item.strip():
+            continue
+        key, value = item.split(':')
+        aliases[key.strip()] = value.strip()
+    return aliases
+
+
+class DownloadError(Exception):
+    """when downloading a network resource fails"""
+
+
+@with_postgres_transactions()
+@with_single_postgres_transaction()
+class FeaturedVersionsAutomaticCronApp(BaseCronApp):
+    app_name = 'featured-versions-automatic'
+    app_description = """Use https://product-details.mozilla.org/1.0/
+    to automatically figure out what product versions should be
+    set as featured.
+    """
+
+    required_config = Namespace()
+    required_config.add_option(
+        'api_endpoint_url',
+        default=(
+            'https://product-details.mozilla.org/1.0/'
+        ),
+        doc='URL from which we can download all the featured versions'
+    )
+    required_config.add_option(
+        'products',
+        default='firefox,mobile,thunderbird',
+        from_string_converter=lambda line: tuple(
+            [x.strip() for x in line.split(',') if x.strip()]
+        ),
+        doc='a comma-delimited list of products to recognize'
+    )
+    required_config.add_option(
+        'aliases',
+        default='mobile:FennecAndroid',
+        from_string_converter=alias_list_to_dict,
+        doc=(
+            'a comma-delimited list of name:RealName. If a product '
+            'does not need an alias match it by capitalizing the name '
+            '(e.g. "thunderbird" -> "Thunderbird")'
+        )
+    )
+
+    def run(self, connection):
+        assert isinstance(self.config.api_endpoint_url, basestring)
+        assert self.config.api_endpoint_url.endswith('/')
+
+        # The @with_single_postgres_transaction decorator makes
+        # sure this cursor is committed or rolled back and cleaned up.
+        cursor = connection.cursor()
+
+        for product in self.config.products:
+            url = '{}{}_versions.json'.format(
```

Comment:
I don't like that. I think it misses the point of configuration. As I see it, it is quite likely that were the server to change, that file path could change as well. 

How about making the config option the entire URL with a parameter? Something like this: 

```python
required_config.add_option(
    'api_endpoint_url',
    default='https://product-details.mozilla.org/1.0/{product}_versions.json',
)
```

And then here you can do: 

```python
url = self.config.api_url.format(product=product)
```